src/libraries/System.Net.Http/src/System/Net/Http/HttpHandlerDefaults.cs

```diff
@@ -0,0 +1,17 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Threading;
+
+namespace System.Net.Http
+{
+    /// <summary>
+    /// Additional default values used used only in this assembly.
+    /// </summary>
+    internal static partial class HttpHandlerDefaults
+    {
+        public static readonly TimeSpan DefaultKeepAlivePingTimeout = TimeSpan.FromSeconds(20);
+        public static readonly TimeSpan DefaultKeepAlivePingDelay = TimeSpan.FromSeconds(0);
```

Comment:
We should use `TimeSpan.InfiniteTimeout` as the "off" mode.

That said, I feel like it'd be fine to have this default to on. @JamesNK do you know what a reasonable default delay is? would 60 seconds be too long?

Reply:
Java and golang default to off. CCore defaults to 5 minutes.

I think we agreed to have it off by default. I'm happy leaving it off. That avoids the overhead of maintaining the timer and related functionality unless someone opts in.

Reply:
By the way, `KeepAlivePingDelay` and `KeepAlivePingTimeout` are nullable in ASP.NET Core. null equals off.

If you don't want nullable then I agree that `TimeSpan.InfiniteTimeout` should equal off.

Reply:
The default changed to `Timeout.InfiniteTimeSpan`